lines-around-directive has false positive with semicolon-less style.

<!--
This template is for bug reports. If you are reporting a bug, please continue on. If you are here for another reason, please see below:

1. To propose a new rule: http://eslint.org/docs/developer-guide/contributing/new-rules
2. To request a change: http://eslint.org/docs/developer-guide/contributing/changes
3. If you have any questions, please stop by our chatroom: https://gitter.im/eslint/eslint

Note that leaving sections blank will make it difficult for us to troubleshoot and we may have to close the issue.
-->

**Tell us about your environment**
- **ESLint Version:** master (6e9ff08cf8ac9188331fcea7905ce162accefe81)
- **Node Version:** 6.9.1
- **npm Version:** 3.8.9

**What parser (default, Babel-ESLint, etc.) are you using?**
- default

**Please show your full configuration:**
- nothing

**What did you do? Please include the actual source code causing the issue.**

``` bash
$ eslint test.js --no-eslintrc --no-ignore
```

`test.js` is:

``` js
/* eslint lines-around-directive: error */

"use strict"

;[1, 2, 3].forEach(doSomething)
```

**What did you expect to happen?**

No errors since there is an empty line between `"use strict"` and the next statement.

**What actually happened? Please include the actual, raw output from ESLint.**

```
$ eslint test.js --no-eslintrc --no-ignore

C:\Users\t-nagashima.AD\Documents\GitHub\eslint\test.js
  3:1  error  Expected newline after "use strict" directive  lines-around-directive

✖ 1 problem (1 error, 0 warnings)
```

--- a/lib/rules/lines-around-directive.js
+++ b/lib/rules/lines-around-directive.js
@@ -64,14 +64,31 @@ module.exports = {
         }
 
         /**
+        * Gets the last token of a node that is on the same line as the rest of the node.
+        * This will usually be the last token of the node, but it will be the second-to-last token if the node has a trailing
+        * semicolon on a different line.
+        * @param {ASTNode} node A directive node
+        * @returns {Token} The last token of the node on the line
+        */
+        function getLastTokenOnLine(node) {
+            const lastToken = sourceCode.getLastToken(node);
+            const secondToLastToken = sourceCode.getTokenBefore(lastToken);
+
+            return lastToken.type === "Punctuator" && lastToken.value === ";" && lastToken.loc.start.line > secondToLastToken.loc.end.line
+                ? secondToLastToken
+                : lastToken;
+        }
+
+        /**
          * Check if node is followed by a blank newline.
          * @param {ASTNode} node Node to check.
          * @returns {boolean} Whether or not the passed in node is followed by a blank newline.
          */
         function hasNewlineAfter(node) {
-            const tokenAfter = sourceCode.getTokenOrCommentAfter(node);
+            const lastToken = getLastTokenOnLine(node);
+            const tokenAfter = sourceCode.getTokenOrCommentAfter(lastToken);
 
-            return tokenAfter.loc.start.line - node.loc.end.line >= 2;
+            return tokenAfter.loc.start.line - lastToken.loc.end.line >= 2;
         }
 
         /**
@@ -91,10 +108,12 @@ module.exports = {
                     location
                 },
                 fix(fixer) {
+                    const lastToken = getLastTokenOnLine(node);
+
                     if (expected) {
-                        return location === "before" ? fixer.insertTextBefore(node, "\n") : fixer.insertTextAfter(node, "\n");
+                        return location === "before" ? fixer.insertTextBefore(node, "\n") : fixer.insertTextAfter(lastToken, "\n");
                     }
-                    return fixer.removeRange(location === "before" ? [node.range[0] - 1, node.range[0]] : [node.range[1], node.range[1] + 1]);
+                    return fixer.removeRange(location === "before" ? [node.range[0] - 1, node.range[0]] : [lastToken.range[1], lastToken.range[1] + 1]);
                 }
             });
         }

--- a/tests/lib/rules/lines-around-directive.js
+++ b/tests/lib/rules/lines-around-directive.js
@@ -655,6 +655,16 @@ ruleTester.run("lines-around-directive", rule, {
             parserOptions: { ecmaVersion: 6 },
             options: [{ before: "always", after: "never" }]
         },
+
+        // https://github.com/eslint/eslint/issues/7450
+        {
+            code: "'use strict'\n\n;foo();",
+            options: [{ before: "never", after: "always" }]
+        },
+        {
+            code: "'use strict'\n;foo();",
+            options: [{ before: "never", after: "never" }]
+        }
     ],
 
     invalid: [
@@ -1598,6 +1608,27 @@ ruleTester.run("lines-around-directive", rule, {
                 "Expected newline before \"use strict\" directive.",
                 "Unexpected newline after \"use asm\" directive."
             ]
+        },
+
+        // https://github.com/eslint/eslint/issues/7450
+        {
+
+            code: "'use strict'\n\n;foo();",
+            output: "'use strict'\n;foo();",
+            options: [{ before: "never", after: "never" }],
+            errors: ["Unexpected newline after \"use strict\" directive."]
+        },
+        {
+            code: "'use strict'\n;foo();",
+            output: "'use strict'\n\n;foo();",
+            options: [{ before: "never", after: "always" }],
+            errors: ["Expected newline after \"use strict\" directive."]
+        },
+        {
+            code: "'use strict'\n;\nfoo();",
+            output: "'use strict'\n\n;\nfoo();",
+            options: [{ before: "never", after: "always" }],
+            errors: ["Expected newline after \"use strict\" directive."]
         }
     ]
 });

